app/models/author.rb

```diff
@@ -0,0 +1,13 @@
+class Author < ApplicationRecord
+  has_and_belongs_to_many :papers
+
+  validates :name, :presence => true
+
+  def self.search(search)
+    if search
+      where('name LIKE ?', "%#{search}%")
+    else
```

Comment:
Without a default value on line 6, 'search' will always be defined. Changed line 6 to `def self.search(search = nil)`